src/main/java/de/hpi/bpt/argos/storage/dataModel/event/type/StatusUpdatedEventType.java

```diff
@@ -0,0 +1,231 @@
+				new PairImpl<>(CAUSE_EVENT_TYPE_ID_ATTRIBUTE_NAME, statusUpdateTrigger.getTypeId()),
+				new PairImpl<>(ENTITY_ID_ATTRIBUTE_NAME, entity.getId()));
+
+		logger.info(String.format("sending status update event: entity: '%1$s' status: '%2$s' -> '%3$s' : response '%4$s'",
+				entity.getName(),
+				oldStatus,
+				newStatus,
+				feedback.getResponseText()));
+
+		return feedback;
+	}
+
+	/**
+	 * This method creates the statusUpdatedEventType in the database
+	 * @return - the newly created statusUpdateEventType
+	 */
+	private static EventType create() {
+		EventType statusUpdatedEventType = new EventTypeImpl();
+		statusUpdatedEventType.setName(NAME);
+		statusUpdatedEventType.setDeletable(false);
+		statusUpdatedEventType.setShouldBeRegistered(true);
+
+		PersistenceAdapterImpl.getInstance().saveArtifacts(statusUpdatedEventType);
+
+		EventQuery statusUpdatedEventQuery = new EventQueryImpl();
+		statusUpdatedEventQuery.setTypeId(statusUpdatedEventType.getId());
+		statusUpdatedEventQuery.setDescription("default query");
+		statusUpdatedEventQuery.setQuery("SELECT * FROM " + NAME);
+
+		PersistenceAdapterImpl.getInstance().saveArtifacts(statusUpdatedEventQuery);
+
+		List<TypeAttribute> typeAttributes = new ArrayList<>();
+		TypeAttribute timestampAttribute = createTypeAttribute(TIMESTAMP_ATTRIBUTE_NAME, statusUpdatedEventType);
+		typeAttributes.add(timestampAttribute);
+		typeAttributes.add(createTypeAttribute(OLD_STATUS_ATTRIBUTE_NAME, statusUpdatedEventType));
+		typeAttributes.add(createTypeAttribute(NEW_STATUS_ATTRIBUTE_NAME, statusUpdatedEventType));
+		typeAttributes.add(createTypeAttribute(CAUSE_EVENT_ID_ATTRIBUTE_NAME, statusUpdatedEventType));
+		typeAttributes.add(createTypeAttribute(CAUSE_EVENT_TYPE_ID_ATTRIBUTE_NAME, statusUpdatedEventType));
+		typeAttributes.add(createTypeAttribute(ENTITY_ID_ATTRIBUTE_NAME, statusUpdatedEventType));
+
+		PersistenceAdapterImpl.getInstance().saveArtifacts(typeAttributes.toArray(new TypeAttribute[typeAttributes.size()]));
+
+		statusUpdatedEventType.setTimeStampAttributeId(timestampAttribute.getId());
+		PersistenceAdapterImpl.getInstance().saveArtifacts(statusUpdatedEventType);
+
+		return statusUpdatedEventType;
+	}
+
+	/**
+	 * This method returns a new typeAttribute for a given eventType.
+	 * @param name - the name of the typeAttribute to create
+	 * @param eventType - the eventType this attribute is for
+	 * @return - the newly created typeAttribute
+	 */
+	private static TypeAttribute createTypeAttribute(String name, EventType eventType) {
+		TypeAttribute typeAttribute = new TypeAttributeImpl();
+		typeAttribute.setName(name);
+		typeAttribute.setTypeId(eventType.getId());
+		return typeAttribute;
+	}
+
+	/**
+	 * This method registers a new eventEntityMapping in the given argos instance.
+	 * @param argos - the argos instance to use for mapping
+	 */
+	private static void registerCustomMapping(Argos argos) {
+		EventCreationObserver customMapping = new EventCreationObserver() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 23 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1188)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1604)